### PR TITLE
loosen check pow 10 test

### PR DESCRIPTION
**Pull request by thedavidmeister**
## Checks

By submitting this for review, I'm confirming I've done the following:
- [ ] made this PR as small as possible
- [ ] unit-tested any new functionality
- [ ] linked any relevant issues or PRs
- [ ] included screenshots (if this involves a front-end change)

## Summary by CodeRabbit

- Documentation
  - Clarified behavior of decimal exponent handling and return values for improved readability.
  - Added explanatory comments to mantissa processing to aid understanding.
  - No functional changes.

- Tests
  - Updated input bounds to cover a wider numeric range and higher exponents.
  - Strengthened assumptions to improve robustness and edge-case coverage.
  - No changes to public APIs or runtime behavior.

### CODE REVIEW

File: src/lib/implementation/LibDecimalFloatImplementation.sol [hunk 1]
```diff
@@ -1008,6 +1008,7 @@ library LibDecimalFloatImplementation {
         }
     }
 
+    /// First 4 digits of the mantissa and whether we need to interpolate.
```
[the remark this answers is not in the document]
Comment (thedavidmeister) [reply src/lib/implementation/LibDecimalFloatImplementation.sol:1011]: it's not helpful to just rephrase the code as text because then later when we change the code the comments are fragile
